api/src/backend/api/tests/test_views.py

```diff
@@ -5458,3 +5458,331 @@ def test_integrations_filters_invalid(self, authenticated_client, filter_name):
+    ):
+        config_id = lighthouse_config_fixture.id
+        response = authenticated_client.delete(
+            reverse("lighthouseconfig-detail", kwargs={"pk": config_id})
+        )
+        assert response.status_code == status.HTTP_204_NO_CONTENT
+        # Verify deletion
+        response = authenticated_client.get(
+            reverse("lighthouseconfig-detail", kwargs={"pk": config_id})
+        )
+        assert response.status_code == status.HTTP_404_NOT_FOUND
+
+    @patch("api.v1.views.openai.OpenAI")
+    def test_lighthouse_config_check_connection(
+        self, mock_openai, authenticated_client, lighthouse_config_fixture
+    ):
+        config_id = lighthouse_config_fixture.id
+        # Mock successful API call
+        mock_client = Mock()
+        mock_client.models.list.return_value = Mock(
+            data=[Mock(id="gpt-4o"), Mock(id="gpt-4o-mini")]
+        )
+        mock_openai.return_value = mock_client
+        # Check connection
+        response = authenticated_client.get(
+            reverse("lighthouseconfig-check-connection", kwargs={"pk": config_id})
+        )
+        assert response.status_code == status.HTTP_200_OK
+        assert response.json()["data"]["detail"] == "Connection successful!"
+        assert "gpt-4o" in response.json()["data"]["available_models"]
+        assert "gpt-4o-mini" in response.json()["data"]["available_models"]
+
+    def test_lighthouse_config_get_key(
+        self, authenticated_client, lighthouse_config_fixture, valid_config_payload
+    ):
+        config_id = lighthouse_config_fixture.id
+        expected_api_key = valid_config_payload["data"]["attributes"]["api_key"]
+        response = authenticated_client.get(
+            reverse("lighthouseconfig-detail", kwargs={"pk": config_id})
+            + "?fields[lighthouse-config]=api_key"
+        )
+        assert response.status_code == status.HTTP_200_OK
+        assert response.json()["data"]["attributes"]["api_key"] == expected_api_key
+
+    def test_lighthouse_config_filters(
+        self, authenticated_client, lighthouse_config_fixture
+    ):
+        # Test name filter
+        response = authenticated_client.get(
+            reverse("lighthouseconfig-list")
+            + "?filter[name]="
+            + lighthouse_config_fixture.name
+        )
+        assert response.status_code == status.HTTP_200_OK
+        assert len(response.json()["data"]) == 1
+        # Test model filter
+        response = authenticated_client.get(
+            reverse("lighthouseconfig-list") + "?filter[model]=gpt-4o"
+        )
+        assert response.status_code == status.HTTP_200_OK
+        assert len(response.json()["data"]) == 1
+        # Test is_active filter
+        response = authenticated_client.get(
+            reverse("lighthouseconfig-list") + "?filter[is_active]=true"
+        )
+        assert response.status_code == status.HTTP_200_OK
+        assert len(response.json()["data"]) == 1
```

Comment:
Use pytest parametrization here, please. Easier to read, maintain and add new tests if required.